src/els_utils.erl

```diff
@@ -66,6 +68,25 @@ start_epmd() ->
     [] = os:cmd(epmd_path() ++ " -daemon"),
     ok.
 
+%% @doc Returns a project-relative file path for a given URI
+-spec project_relative(uri()) -> file:filename().
+project_relative(Uri) ->
+  Path = els_uri:path(Uri),
+
+  RootDir = els_config:get(root_uri),
+  RootList = filename:split(els_uri:path(RootDir)),
+  PathList = filename:split(Path) ,
+  RelFileBin = filename:join(els_utils:strip_prefix(RootList, PathList)),
+  RelFile = binary_to_list(RelFileBin),
+  RelFile.
+
+%% @doc Strips the prefix in List1 from List2
```

Comment:
Isn't there an OTP API for this?

Reply:
I looked for one, without success. If there is one, I will gladly use it.

Reply:
`string:prefix/2` removes the prefix if it is there otherwise returns `nomatch`.

Reply:
Thanks. And I presume we don't have to worry about path separators, and platform issues? As we will always be working with the same file system while doing this.

Reply:
That is a good point, but as you say both the file's URI and the root's URI have the same format since they are defined based on the same filesystem. We are using `els_uri:path/1` which just strips out leading `file://` prefix, so I think we can safely rely on `string:prefix/2` to work, unless I'm missing something 😄